### PR TITLE
Add compound to `lending.borrow`

### PR DESCRIPTION
**Pull request by tomfutago**

### CODE REVIEW

File: models/_sector/lending/borrow/ethereum/platforms/exposed/compound_v2_ethereum_borrow.sql
Comment (jeff-dude) [reply models/_sector/lending/borrow/ethereum/platforms/exposed/compound_v2_ethereum_borrow.sql:1]: this is an important decision, thinking deeper about it. it'll help set the framework for how we do this on all sector spells / project views on top. i'm leaning towards the above, where exposed views are back in project directory. curious what you think (also cc @Hosuke for opinion)?

to be clear:
- all spells that help build final sector spell live in the `models/_sector/...` directory path
- all spells that sit on top go back into their project directory `models/<project>/...`

**Review by jeff-dude — CHANGES_REQUESTED** [review]
fantastic work here 🔥  loving how these sector spells come together

few minor comments below to address prior to merge

File: models/_sector/lending/borrow/ethereum/platforms/exposed/compound_v2_ethereum_borrow.sql
Comment (Hosuke) [reply models/_sector/lending/borrow/ethereum/platforms/exposed/compound_v2_ethereum_borrow.sql:1]: I agree with the proposed structure. 

Placing 
- sector-specific spells in `models/_sector/...` 
- and project-specific spells in `models/<project>/...` 

provides a clear, organized approach that enhances both navigability and maintainability. 

To aid in defining what constitutes a sector spell versus a project spell, we could utilize the `post_hook='{{ expose_spells }}` setting. (Those spell with exposed views can be put under project directory) This tag may serve as a clear indicator, helping to guide the placement of spells in the appropriate directory.

Comment (tomfutago) [reply models/_sector/lending/borrow/ethereum/platforms/exposed/compound_v2_ethereum_borrow.sql:1]: I agree too, thanks both 🙏 With my "exposed" folders I wanted to keep all project related spells all in sector location, but as with Compound/Balancer example - there are also cross-sector spells, so splitting per sector and project makes sense 💯 Will go ahead and make those changes.

**Comment by tomfutago** [comment]
Update summary:
* exposed views moved back to their respective project folders
* in previous version I had `ctokens` hardcoded as part of individual spell, but they are actually needed for `supply` and other spells, so I extracted them as standalone static models and put them in project folders too (existing `compound_v2_ethereum_ctokens` spell is exposed in UI, but I left new ones "hidden")
* blank usd amounts issue is still present (similar to `supply` spell [note](https://github.com/duneanalytics/spellbook/pull/4853#issuecomment-1827986727)) - is that linked to your note [here](https://github.com/duneanalytics/spellbook/issues/4759#issuecomment-1845887403) @jeff-dude?

**Review by jeff-dude — APPROVED** [review]

**Comment by jeff-dude** [comment]
> Update summary:
> 
> * exposed views moved back to their respective project folders
> * in previous version I had `ctokens` hardcoded as part of individual spell, but they are actually needed for `supply` and other spells, so I extracted them as standalone static models and put them in project folders too (existing `compound_v2_ethereum_ctokens` spell is exposed in UI, but I left new ones "hidden")
> * blank usd amounts issue is still present (similar to `supply` spell [note](https://github.com/duneanalytics/spellbook/pull/4853#issuecomment-1827986727)) - is that linked to your note [here](https://github.com/duneanalytics/spellbook/issues/4759#issuecomment-1845887403) @jeff-dude?

thanks for the clear info on updates!

as for the usd amount issue, i'm not fully sure tbh. i haven't had a chance to look at your example yet. do you think that's blocking this from being merged (it's ready otherwise, imo)? if blocked, how should we proceed?

**Comment by tomfutago** [comment]
> > Update summary:
> > 
> > * exposed views moved back to their respective project folders
> > * in previous version I had `ctokens` hardcoded as part of individual spell, but they are actually needed for `supply` and other spells, so I extracted them as standalone static models and put them in project folders too (existing `compound_v2_ethereum_ctokens` spell is exposed in UI, but I left new ones "hidden")
> > * blank usd amounts issue is still present (similar to `supply` spell [note](https://github.com/duneanalytics/spellbook/pull/4853#issuecomment-1827986727)) - is that linked to your note [here](https://github.com/duneanalytics/spellbook/issues/4759#issuecomment-1845887403) @jeff-dude?
> 
> thanks for the clear info on updates!
> 
> as for the usd amount issue, i'm not fully sure tbh. i haven't had a chance to look at your example yet. do you think that's blocking this from being merged (it's ready otherwise, imo)? if blocked, how should we proceed?

Blank usd amounts seem like a blocker to me - I mean, that's probably one of the main points of having final data enrichment and for users usd amounts would be the clearest way to compare borrowing volumes etc. 

As to how to proceed, I'm not entirely sure tbh. As I described in the linked note, I'm confused as to why the same code if run manually pulls usd amounts correctly, but not when it runs in CI. I might not be seeing something obvious tho, so another pair of eyes would be definitely helpful 🙏 or, if we can back-populate usd amounts somehow later?

**Comment by jeff-dude** [comment]
> As to how to proceed, I'm not entirely sure tbh. As I described in the linked note, I'm confused as to why the same code if run manually pulls usd amounts correctly, but not when it runs in CI. I might not be seeing something obvious tho, so another pair of eyes would be definitely helpful 🙏 or, if we can back-populate usd amounts somehow later?

ah! i think i finally have the answer:
- enrich macros filter prices on incremental runs, as expected
- enrich macros **_don't_** filter the `base_borrow` / `base_supply` sources on incremental runs
- CI runs history, i assume all `usd_amount` is correct
- CI runs incremental step, since prices is filtered and source is not, all history of amounts is wiped out to null

i bet if we stopped CI mid-run before incremental step, we'd see that. not sure you need to take the time to do that, but feel free if ya want. otherwise, i'd suggest putting the source filters to match and we should be good 🤞 

this should apply to all open PRs in this sector, from what i can tell

**Comment by tomfutago** [comment]
> > As to how to proceed, I'm not entirely sure tbh. As I described in the linked note, I'm confused as to why the same code if run manually pulls usd amounts correctly, but not when it runs in CI. I might not be seeing something obvious tho, so another pair of eyes would be definitely helpful 🙏 or, if we can back-populate usd amounts somehow later?
> 
> ah! i think i finally have the answer:
> 
> * enrich macros filter prices on incremental runs, as expected
> * enrich macros **_don't_** filter the `base_borrow` / `base_supply` sources on incremental runs
> * CI runs history, i assume all `usd_amount` is correct
> * CI runs incremental step, since prices is filtered and source is not, all history of amounts is wiped out to null
> 
> i bet if we stopped CI mid-run before incremental step, we'd see that. not sure you need to take the time to do that, but feel free if ya want. otherwise, i'd suggest putting the source filters to match and we should be good 🤞
> 
> this should apply to all open PRs in this sector, from what i can tell

Woohoo, well spotted, that was it!!! Fixed now 😊 thank you ser! 🙏

**Review by jeff-dude — APPROVED** [review]
lfg 🚀

**Pull request merged**
